File: stepik-union/src/main/java/org/stepik/plugin/StepikPluginConfigurator.java
package org.stepik.plugin;

import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import org.stepik.core.StepikProjectManager;
import org.stepik.core.StudyBasePluginConfigurator;
import org.stepik.plugin.actions.navigation.StepikNextStepAction;
import org.stepik.plugin.actions.navigation.StepikPreviousStepAction;
import org.stepik.plugin.actions.step.DownloadSubmission;
import org.stepik.plugin.actions.step.InsertStepikDirectives;
import org.stepik.plugin.actions.step.OpenInBrowserAction;
import org.stepik.plugin.actions.step.StepikResetStepAction;
import org.stepik.plugin.actions.step.StepikSendAction;
import org.stepik.plugin.actions.step.TestSamplesAction;

public class StepikPluginConfigurator extends StudyBasePluginConfigurator {
    @NotNull
    @Override
    public DefaultActionGroup getActionGroup(Project project) {
        final DefaultActionGroup group = new DefaultActionGroup();

        group.add(new StepikSendAction());
        group.add(new TestSamplesAction());
        group.add(new StepikPreviousStepAction());
        group.add(new StepikNextStepAction());
        group.add(new StepikResetStepAction());
        group.add(new DownloadSubmission());
        group.add(new InsertStepikDirectives());
        group.add(new OpenInBrowserAction());

        return group;
    }

    @Override
    public boolean accept(@NotNull Project project) {
        return StepikProjectManager.isStepikProject(project);
    }
}

File: stepik-union/src/main/kotlin/org/stepik/plugin/StepikPluginConfigurator.kt
package org.stepik.plugin

import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.project.Project
import org.stepik.core.ProjectGenerator
import org.stepik.core.StudyBasePluginConfigurator
import org.stepik.core.StudyUtils.isStepikProject
import org.stepik.core.actions.step.DownloadSubmission
import org.stepik.core.actions.step.InsertStepikDirectives
import org.stepik.core.actions.step.StepikResetStepAction
import org.stepik.core.actions.step.StepikSendAction
import org.stepik.core.actions.step.TestSamplesAction
import org.stepik.core.courseFormat.StepNode
import org.stepik.core.courseFormat.StudyNode
import org.stepik.core.projectWizard.idea.BaseModuleBuilder
import org.stepik.plugin.actions.navigation.StepikNextStepAction
import org.stepik.plugin.actions.navigation.StepikPreviousStepAction
import org.stepik.plugin.actions.navigation.StudyNavigator
import org.stepik.plugin.actions.step.OpenInBrowserAction
import org.stepik.plugin.projectWizard.StepikProjectGenerator
import org.stepik.plugin.projectWizard.idea.SandboxModuleBuilder
import org.stepik.plugin.projectWizard.idea.StepModuleBuilder

class StepikPluginConfigurator : StudyBasePluginConfigurator() {
    override fun getSandboxModuleBuilder(path: String): BaseModuleBuilder {
        return SandboxModuleBuilder(path)
    }

    override fun getStepModuleBuilder(moduleDir: String, step: StepNode): BaseModuleBuilder {
        return StepModuleBuilder(moduleDir, step)
    }

    override fun getProjectGenerator(): ProjectGenerator {
        return StepikProjectGenerator
    }

    override fun nextAction(node: StepNode): StudyNode<*, *>? {
        return StudyNavigator.nextLeaf(node)
    }

    override fun getActionGroup(project: Project): DefaultActionGroup {
        val group = DefaultActionGroup()
        group.addAll(
                StepikSendAction(),
                TestSamplesAction(),
                StepikPreviousStepAction(),
                StepikNextStepAction(),
                StepikResetStepAction(),
                DownloadSubmission(),
                InsertStepikDirectives(),
                OpenInBrowserAction()
        )

        return group
    }

    override fun accept(project: Project): Boolean {
        return isStepikProject(project)
    }
}
